File: software/examples/c/example_morse_code.c

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_piezo_buzzer.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main() {
	// Create IP connection to brickd
	IPConnection ipcon;
	if(ipcon_create(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not create connection\n");
		exit(1);
	}

	// Create device object
	PiezoBuzzer pb;
	piezo_buzzer_create(&pb, UID); 

	// Add device to IP connection
	if(ipcon_add_device(&ipcon, &pb) < 0) {
		fprintf(stderr, "Could not connect to Bricklet\n");
		exit(1);
	}
	// Don't use device before it is added to a connection

	// Morse SOS
	piezo_buzzer_morse_code(&pb, "... --- ...");

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon);
}

File: software/examples/go/example_morse_code.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
	"github.com/tinkerforge/go-api-bindings/piezo_buzzer_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Piezo Buzzer Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	pb, _ := piezo_buzzer_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Morse SOS
	pb.MorseCode("... --- ...")

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
